### PR TITLE
humioctl: init at 0.25.0

### PR DESCRIPTION
**Pull request by lucperkins**
###### Motivation for this change

This PR adds the CLI for [Humio](https://www.humio.com/), `humioctl`, to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lucperkins** [comment]
@kalbasit Any chance I can get a +1 on this? :)

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4669,6 +4669,12 @@
       fingerprint = "5B93 9CFA E8FC 4D8F E07A  3AEA DFE1 D4A0 1733 7E2A";
     }];
   };
+  lucperkins = {
+    email = "lucperkins@gmail.com";
+    github = "lucperkins";
+    githubId = 1523104;
+    name = "Luc Perkins";
+  };
```
Comment (kalbasit) [inline maintainers/maintainer-list.nix:4677]: This should be in a separate commit:


`maintainers: add lucperkins`

File: pkgs/applications/logging/humioctl/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ buildGoModule, fetchFromGitHub, installShellFiles, stdenv }:
+
+let
+  version = "0.25.0";
+  sha256 = "1x8354m410nf9g167v0i1c77s5w2by7smdlyjwl89ixgdjw04ay3";
+in buildGoModule {
```
Comment (kalbasit) [inline pkgs/applications/logging/humioctl/default.nix:6]: Please inline these, there's no point to define these as local variables. You'll have to use `rec` in order to define and use the version in the same derivation FYI.

File: pkgs/applications/logging/humioctl/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    name = "humioctl";
+    pname = "humioctl";
+    version = version;
+
+    goPackagePath = "github.com/humio/cli";
```
Comment (kalbasit) [inline pkgs/applications/logging/humioctl/default.nix:11]:
```suggestion
```

This is not used by buildGoModule.

File: pkgs/applications/logging/humioctl/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    vendorSha256 = "14bysjgvahr56hvd8walym11hh721i1q2g503n8m68wdzrrym4qy";
+
+    src = fetchFromGitHub {
+      owner = "humio";
+      repo = "cli";
+      rev = "v${version}";
+      sha256 = sha256;
+    };
+
+    subPackages = ["."];
```
Comment (kalbasit) [inline pkgs/applications/logging/humioctl/default.nix:22]:
```suggestion
```

I don't believe that you need this, but I could be wrong.

**Review by kalbasit — CHANGES_REQUESTED** [review]
Please squash all the commits regarding humioctl into one: `humioctl: init at 0.25.0`

File: pkgs/applications/logging/humioctl/default.nix (hunk 2 above)
Comment (lucperkins) [reply pkgs/applications/logging/humioctl/default.nix]: I like having these at the top so that it's immediately clear which variables need to be updated for each update. Furthermore, I've come to see `rec` as an [anti-pattern](https://nix.dev/anti-patterns/language.html#rec-expression). I have, however, made the other suggested changes.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build humioctl

**Review by kalbasit — APPROVED** [review]
